Zend/zend_API.c

```diff
@@ -2158,7 +2158,7 @@ ZEND_API int zend_register_functions(zend_class_entry *scope, const zend_functio
 		internal_function->prototype = NULL;
 		if (ptr->flags) {
 			if (!(ptr->flags & ZEND_ACC_PPP_MASK)) {
-				if (ptr->flags != ZEND_ACC_DEPRECATED || scope) {
+				if (ptr->flags != ZEND_ACC_DEPRECATED && scope) {
```

Comment:
This change looks unrelated but I asked Nikita about it and he said this:

> there was a bug elsewhere preventing me from using ZEND_ACC_DEPRECATED in ZEND_ME without also specifying ZEND_ACC_PUBLIC

(tl;dr is related)